src/Acuminator/Acuminator.Analyzers/StaticAnalysis/CallingBaseDataViewDelegate/CallingBaseDataViewDelegateFromOverrideDelegateAnalyzer.cs

```diff
@@ -0,0 +1,117 @@
+            var ownRelatedViews = pxGraph.Views
+                                  .Where(v => ownDelegatesDictionary.ContainsKey(v.Symbol.Name) &&
+                                              pxGraph.Symbol.Equals(v.Symbol.ContainingType))
+                                  .ToArray();
+            var baseNonRedeclaredRelatedViews = new List<DataViewInfo>();
+
+            foreach (var view in pxGraph.Views)
+            {
+                for (var curView = view; curView != null; curView = curView.Base)
+                {
+                    if (ownDelegatesDictionary.ContainsKey(curView.Symbol.Name) &&
+                        !pxGraph.Symbol.Equals(curView.Symbol.ContainingType) &&
+                        ownRelatedViews.All(v => !v.Symbol.Name.Equals(curView.Symbol.Name, StringComparison.OrdinalIgnoreCase)))
+                    {
+                        baseNonRedeclaredRelatedViews.Add(curView);
+                    }
+                }
+            }
+
+            foreach (var viewDelegate in ownDelegatesDictionary.Values)
+            {
+                var walker = new Walker(context, pxContext, baseNonRedeclaredRelatedViews);
+
+                walker.Visit(viewDelegate.Node);
+            }
+        }
+
+        private class Walker : NestedInvocationWalker
+        {
+            private readonly SymbolAnalysisContext _context;
+            private readonly PXContext _pxContext;
+            private readonly IEnumerable<DataViewInfo> _nonRedeclaredBaseViews;
+
+            public Walker(SymbolAnalysisContext context, PXContext pxContext, IEnumerable<DataViewInfo> nonRedeclaredBaseViews)
+                : base(context.Compilation, context.CancellationToken)
+            {
+                pxContext.ThrowOnNull();
+                nonRedeclaredBaseViews.ThrowOnNull();
+
+                _context = context;
+                _pxContext = pxContext;
+                _nonRedeclaredBaseViews = nonRedeclaredBaseViews;
+            }
+
+            public override void VisitMemberAccessExpression(MemberAccessExpressionSyntax node)
+            {
+                ThrowIfCancellationRequested();
+                _context.CancellationToken.ThrowIfCancellationRequested();
+
+                var reported = false;
+                var symbol = GetSymbol<ISymbol>(node);
+                var expressionSymbol = GetSymbol<ISymbol>(node.Expression);
+
+                //Base.PXSelectBaseGenIns.Select()
+                if (_pxContext.PXSelectBaseGeneric.Select.Contains(symbol.OriginalDefinition))
+                {
+                    reported = TryToReport(expressionSymbol, node);
+                }
+                //Base.PXSelectBaseGenIns.View.Select()
+                else if (_pxContext.PXView.Select.Contains(symbol) &&
+                         _pxContext.PXSelectBase.View.Equals(expressionSymbol) &&
+                         node.Expression is MemberAccessExpressionSyntax expressionNode)
```

Comment:
@metallmonster there is no support for ConditionalMemberAccess like this:
```
Base.PXSelectBaseGenIns?.View?.Select()
```
On the other hand nobody probably will write such code. Only if they want to skip diagnostic.

Reply:
ok